example/http/request.cpp

```diff
@@ -68,24 +107,26 @@ int main(int argc, char **argv) {
     }
     settings["http/url"] = argv[0];
 
-    loop_with_initial_event([&]() {
-        http::request(
-            settings,
-            headers,
-            body,
-            [](Error error, Var<http::Response> response) {
-                if (error) {
-                    std::cout << "Error: " << error.explain() << "\n";
-                    break_loop();
-                    return;
-                }
-                std::cout << response->response_line << "\n";
-                for (auto &pair : response->headers) {
-                    std::cout << pair.first << ": " << pair.second << "\n";
-                }
-                std::cout << "\n" << response->body << "\n";
-                break_loop();
-            });
+    loop_with_initial_event([=]() {
+        http::request_connect(settings, [=](Error e, Var<net::Transport> t) {
+            if (e) {
+                throw e;
+            }
+            http::request_send(
+                t, settings, headers, body, [=](Error e, Var<http::Request> r) {
+                    if (e) {
+                        throw e;
+                    }
+                    mk::warn("Request sent... waiting for response...");
+                    t->on_data(fcompose(http_read_response_coro(),
+                                        fcompose(add_request(std::move(r)),
+                                                 print_response_and_exit())));
```

Comment:
With more template-foo, `fcompose` will be able to:

1. chain more than two functors at once

2. deduce the argument types of the first functor and apply them, so to avoid the need to pass the first argument as a `std::function<>` explicitly, which is very annoying

So, this piece of code should actually be imagined as:

```C++
t->on_data(fcompose(
    http_read_response_coro(),
    [r = std::move(r)](http::Response x) {
        x.request = r;
        return x;
    },
    [=](http::Response resp) {
        // Here code of `print_response_and_exit`
    });
```